### ISSUE TITLE
feat: changing tab leads to tab's root state

### ISSUE THREAD

**Issue by Buffaloketchup** [driftyco/ionic#3705]
**Type**: <span ionic-type>feat</span>

**Platform**: <span ionic-platform>all</span>  

<span ionic-description>On a Ionic app using Angular's $stateprovider with a state tree of root states and sub-states, and the "ion-tab" directive, selecting a tab that has already been visited will bring the user to the last sub-state he visited. You need to tap a second time in order to go back to the root state.

It's probably the desired effect for most of the apps, however in certain cases you might want to force the app the always redirect to the root state of a tab when selecting it.

In my case for example, the root state holds a controller that loads data from a server and this data is used in all of its sub-states. But actions done in other tabs might modify the database and so, when the user gets back to the first tab, it is necessary to update the data.

I've been looking over the internet and found several hacks for this but nothing very handy especially when dealing with many tabs and sub-states.

It'd be very handy to have a built-in option, maybe in the "ion-tab-bar" or the "ion-tab", to force redirection to the root state of a tab whenever that tab is selected.</span>

<span is-issue-template></span>

**Comment by Ionitron** [comment]
Greetings @Buffaloketchup!

My sensors indicate that you need to update your issue through our [custom issue form](http://ionicframework.com/submit-issue/#/?iid=3705). We are now requiring all issues to be submitted this way, to ensure that we have all of the information necessary to fix them as quickly as possible.

**[Click Here To Update Your Issue](http://ionicframework.com/submit-issue/#/?iid=3705)**

I will have no choice but to close this issue if it is not resubmitted through the form. Please fill out the rest of the form, so that I may use my friendly robot powers to assist you.

Thank you!

<span ionitron-issue-resubmit></span>

**Comment by perrygovier** [comment]
I suspect you want to take a look at the navigation events that are broadcasted.
http://ionicframework.com/docs/api/directive/ionView/

Check the situation before enter and redirect accordingly.

**Comment by Buffaloketchup** [comment]
I read through that page but didn't find any help.

To make think a bit clearer, here's a sample of my html tab-bar:

    <ion-tab title="Accounts" href="#/tab/accounts">
        <ion-nav-view name="tab-accounts"></ion-nav-view>
    </ion-tab>

In that case, despite the "href" pointing to "#/tab/accounts", if I navigate to, let's say "#/tab/accounts/sub-account1", then go to another tab and back to the "account" tab, I will be brought back to the "sub-account1" state instead of "accounts" root state.

**Comment by perrygovier** [comment]
Depending on the details of the app, there's probably a few ways to fix the issue.

By default tabs has parallel history stacks, so as you switch between tabs, they have their own history and idea of where "back" takes you. Switching between them, you don't automatically go to the root of that history stack. You go to the currently active page in that stack. 

If this isn't desired, you can normally adjust `$ionicHistory` to your liking.

You may be able to use the navigation events to tell the page to update its data on the detail page accordingly. (my favorite option)

If the tabs history logic isn't something you want, there's nothing wrong with not using them. The visual style can be recreated using `div`s with the right classes. 


This might be more of a "How do I" type of question that's better for the forums. Do any of those approaches work?

**Comment by Buffaloketchup** [comment]
Thanks for your suggestions. I did ask the question on the ionic forums 2 weeks ago but never got a satisfying reply, so I thought there was no good way of doing it. That's why I came to ask here.

But I actually found a solution. I replaced the "href" attribute with a "ng-click" calling a function. That function simply uses a $state.go method pointing to the root state of every tab.  So I don't have to deal with $ionicHistory and all that.

Don't know why I haven't thought of it before.

Anyway thank you perrygovier for your help.
